kurl_proxy/cmd/main.go

```diff
@@ -430,26 +434,25 @@ func getHttpsServer(upstream, dexUpstream *url.URL, tlsSecretName string, secret
 			}
 		}()
 	})
-	mux.Handle("/tls", r)
-	mux.Handle("/tls/", r)
-
-	// mux.Handle("/api/v1/kots/", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-	// 	log.Println("Kots REST API not proxied.")
-	// 	http.Error(w, "Not found", http.StatusNotFound)
-	// }))
 
 	if dexUpstream != nil {
-		dexReverseProxy := httputil.NewSingleHostReverseProxy(dexUpstream)
-		mux.Handle("/dex", dexReverseProxy)
-		mux.Handle("/dex/", dexReverseProxy)
+		r.Any("/dex/*path", gin.WrapH(httputil.NewSingleHostReverseProxy(dexUpstream)))
```

Comment:
does this work with just `/dex` as a path?

Reply:
not exactly sure what is served from `/dex`, but here’s a curl output:
```
curl -skIL https://34.85.163.117:8800/dex
HTTP/1.1 307 Temporary Redirect
Content-Type: text/html; charset=utf-8
Location: /dex/
Date: Thu, 09 Mar 2023 16:43:15 GMT

HTTP/1.1 404 Not Found
Content-Length: 19
Content-Security-Policy: frame-ancestors 'none';
Content-Type: text/plain; charset=utf-8
Date: Thu, 09 Mar 2023 16:43:15 GMT
X-Content-Type-Options: nosniff
X-Frame-Options: DENY
```

it handles the request and redirects to `/dex/`, but then that returns 404 ([which seems expected](https://github.com/dexidp/dex/blob/master/server/server.go#L354-L395))

Reply:
Sharing another finding here, it seems that it is gin that is doing the redirect above, but making requests to the clusterip of the dex service directly shows that both endpoint would return 404
```
$ curl -skIL http://10.96.1.133:5556/dex
HTTP/1.1 404 Not Found
Content-Type: text/plain; charset=utf-8
X-Content-Type-Options: nosniff
Date: Thu, 09 Mar 2023 17:02:17 GMT
Content-Length: 19

$ curl -skIL http://10.96.1.133:5556/dex/
HTTP/1.1 404 Not Found
Content-Type: text/plain; charset=utf-8
X-Content-Type-Options: nosniff
Date: Thu, 09 Mar 2023 17:02:21 GMT
Content-Length: 19
```